powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/service/model/signature/SignatureMetadataConverterTest.java

```diff
@@ -0,0 +1,99 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class for SignatureMetadataConverter.
+ * This class tests various aspects of converting SignatureMetadata to and from its serialized JSON form.
+ *
+ * @author Jan Dusil
+ */
+public class SignatureMetadataConverterTest {
+
+    /**
+     * Converter object to be used for tests.
+     */
+    private SignatureMetadataConverter converter;
+
+    /**
+     * Initializes the SignatureMetadataConverter object and any other necessary objects.
+     */
+    @BeforeEach
+    public void setUp() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        converter = new SignatureMetadataConverter(objectMapper);
+    }
+
+    /**
+     * Tests the conversion of a PowerAuthSignatureMetadata object to its serialized JSON form.
+     */
+    @Test
+    public void convertToDatabaseColumnTest() {
+        PowerAuthSignatureMetadata metadata = new PowerAuthSignatureMetadata("POST", "123");
+        String jsonStr = converter.convertToDatabaseColumn(metadata);
+
+        assertNotNull(jsonStr);
+        assertEquals("{\"type\":\"PowerAuthSignatureMetadata\",\"signatureDataMethod\":\"POST\",\"signatureDataUriId\":\"123\"}", jsonStr);
+    }
+
+    /**
+     * Tests the conversion of a serialized JSON string back to a SignatureMetadata object.
+     */
+    @Test
+    public void convertToEntityAttributeTest() {
+        String jsonStr = "{\"type\":\"PowerAuthSignatureMetadata\",\"signatureDataMethod\":\"POST\",\"signatureDataUriId\":\"123\"}";
+        SignatureMetadata<?, ?> metadata = converter.convertToEntityAttribute(jsonStr);
+
+        assertNotNull(metadata);
+        assertEquals("POST", metadata.getMetadataParam1());
+        assertEquals("123", metadata.getMetadataParam2());
+    }
+
+    /**
+     * Tests a round-trip conversion, from object to JSON string and back to object, to ensure consistency.
+     */
+    @Test
+    public void testRoundTripConversion() {
+        PowerAuthSignatureMetadata originalMetadata = new PowerAuthSignatureMetadata("POST", "123");
+        String jsonStr = converter.convertToDatabaseColumn(originalMetadata);
+        PowerAuthSignatureMetadata convertedMetadata = (PowerAuthSignatureMetadata) converter.convertToEntityAttribute(jsonStr);
+
+        assertNotNull(convertedMetadata);
+        assertEquals(originalMetadata.getMetadataParam1(), convertedMetadata.getMetadataParam1());
+        assertEquals(originalMetadata.getMetadataParam2(), convertedMetadata.getMetadataParam2());
+    }
+
+    /**
+     * Tests the converter's behavior when provided with an invalid JSON string.
+     */
+    @Test
+    public void testInvalidJsonInput() {
+        String invalidJson = "{\"invalidField\":\"someValue\"}";
+        SignatureMetadata<?, ?> metadata = converter.convertToEntityAttribute(invalidJson);
```

Comment:
🤔 I know that Java generics are tricky. But don't we have anything better than unbounded wildcard type?